hashicorp/dev-portal
# fix: address MDX render issues

## ✅ "Ready for Review" checklist


---

## 🔗 Relevant links

- 🔎 [Preview - /waypoint/tutorials/deploy-aws/aws-lambda][/waypoint/tutorials/deploy-aws/aws-lambda] 
- 🔎 [Preview - /vault/tutorials/secrets-management/multi-factor-authentication][/vault/tutorials/secrets-management/multi-factor-authentication]
- 🎟️ [[Tutorial Content] Codeblock, tabs bug with image border](https://app.asana.com/0/1202001387788447/1202286147442117/f)
- 🎟️ [[Tutorial Content] Alert renders dark like a codeblock?](https://app.asana.com/0/1202001387788447/1202286147442119/f)
- 🎟️ [Tabs rendering in an image container for some reason](https://app.asana.com/0/1202097197789424/1202221422987978/f)

## 🗒️ What

Fixes an issue where the MDX custom `img` component seems to cause a variety of hydration mismatch issues.

## 🛠️ How

- Replaces a `div` with a `span`

## 🧪 Testing

- Visit [/waypoint/tutorials/deploy-aws/aws-lambda][/waypoint/tutorials/deploy-aws/aws-lambda], and ensure the page looks as expected
    - The code block at the start of the `Update lambda function` section should look like a code block, not an image
    - The blue alert component at the end of the `Update lamba function` section should look like a blue alert box, not a code block
- Visit [/vault/tutorials/secrets-management/multi-factor-authentication][/vault/tutorials/secrets-management/multi-factor-authentication]
    - The tabs near the start of the `Scenario tasks` section should render as expected (previously they rendered into an image container)

## 💭 Anything else?

I think (hope!) this fix may solve a number of MDX rendering issues we've seen throughout `/tutorials` routes - and possibly in `/docs` routes as well. 

[/waypoint/tutorials/deploy-aws/aws-lambda]: https://dev-portal-git-zsaddress-mdx-render-issues-hashicorp.vercel.app/waypoint/tutorials/deploy-aws/aws-lambda
[/vault/tutorials/secrets-management/multi-factor-authentication]: https://dev-portal-git-zsaddress-mdx-render-issues-hashicorp.vercel.app/vault/tutorials/secrets-management/multi-factor-authentication
